haystack/components/generators/chat/openai.py

```diff
@@ -558,6 +549,10 @@ def _convert_chat_completion_chunk_to_streaming_chunk(self, chunk: ChatCompletio
         :returns:
             The StreamingChunk.
         """
+        # if there are no choices, return an empty chunk
+        if len(chunk.choices) == 0:
+            return StreamingChunk(content="", meta={"model": chunk.model, "received_at": datetime.now().isoformat()})
```

Comment:
I added this to handle the case where no choices are present in a streaming chunk (e.g. like in the usage chunk)